product_supplierinfo_for_customer_sale/models/product_product.py

```diff
@@ -13,6 +13,8 @@ class ProductProduct(models.Model):
     def name_search(self, name='', args=None, operator='ilike', limit=80):
         res = super(ProductProduct, self).name_search(
             name, args=args, operator=operator, limit=limit)
+        if not limit or len(res) >= limit:
+            limit = (limit - len(res)) if limit else False
```

Comment:
```suggestion
            limit = (limit - len(res)) if limit else sys.maxsize
```
importing `sys` previously